### DISCUSSION TITLE
Where can I find the debian package?

### DISCUSSION THREAD

**Discussion by chmduquesne** [TimWolla/haproxy-auth-request#38] [Q&A]
Hi there,

After reading the [article](https://bl.duesterhus.eu/20180119/), I was under the impression that I would be able to find a debian package. Since I can't find it via apt, I am assuming that it is available in a non-official repository, or at least as a binary .deb somewhere.

Where can I find it, if it exists?

Cheers,
Christophe-Marie

**Comment by TimWolla** [comment]
> Where can I find it, if it exists?

It doesn't. You will need to build it yourself.

**Reply by chmduquesne** [reply]
Hi!

Thank you. I am quite new to debian packaging. What would be the commands to run? In which directory? I tried

    dh_make --native -p haproxy-auth-request_1.0.0

but it says that the debian directory already exists.

Thanks!

**Reply by TimWolla** [reply]
`dpkg-buildpackage` should do everything you need.

**Reply by chmduquesne** [reply]
Thanks! FYI I gave it a try, it gives me the following error:

```
$ dpkg-buildpackage       
dpkg-buildpackage: info: source package haproxy-auth-request
dpkg-buildpackage: info: source version 0
dpkg-buildpackage: info: source distribution UNRELEASED
dpkg-buildpackage: info: source changed by Tim Düsterhus <tim@bastelstu.be>
dpkg-buildpackage: info: host architecture amd64
 dpkg-source --before-build .
 fakeroot debian/rules clean
dh clean
   dh_auto_clean
   dh_clean
 dpkg-source -b .
dpkg-source: info: using source format '3.0 (native)'
dpkg-source: info: building haproxy-auth-request in haproxy-auth-request_0.tar.xz
dpkg-source: info: building haproxy-auth-request in haproxy-auth-request_0.dsc
 debian/rules build
dh build
   dh_update_autotools_config
   dh_auto_configure
   dh_auto_build
	make -j1
make[1]: Entering directory '/home/chmd/code/selfcompiled/haproxy-auth-request'
make[1]: Nothing to be done for 'default'.
make[1]: Leaving directory '/home/chmd/code/selfcompiled/haproxy-auth-request'
   dh_auto_test
 fakeroot debian/rules binary
dh binary
   dh_testroot
   dh_prep
   dh_auto_install
	make -j1 install DESTDIR=/home/chmd/code/selfcompiled/haproxy-auth-request/debian/haproxy-auth-request AM_UPDATE_INFO_DIR=no
make[1]: Entering directory '/home/chmd/code/selfcompiled/haproxy-auth-request'
make[1]: *** No rule to make target 'haproxy-lua-http/http.lua', needed by 'install'.  Stop.
make[1]: Leaving directory '/home/chmd/code/selfcompiled/haproxy-auth-request'
dh_auto_install: error: make -j1 install DESTDIR=/home/chmd/code/selfcompiled/haproxy-auth-request/debian/haproxy-auth-request AM_UPDATE_INFO_DIR=no returned exit code 2
make: *** [debian/rules:7: binary] Error 2

```

I could not figure out how to debug this, so I thought I would at least let you know.

Cheers,
Christophe-Marie

**Reply by TimWolla** [reply]
> I could not figure out how to debug this, so I thought I would at least let you know.

You'll need to initialize the `haproxy-lua-http` submodule, as it will be included within the same Debian package (technically violating packaging rules by bundling libraries):

- `git submodule init`
- `git submodule update`
